src/VisualStudio/Next/Remote/JsonRpcClient.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using StreamJsonRpc;
+using Microsoft.CodeAnalysis.Remote;
+
+namespace Microsoft.VisualStudio.LanguageServices.Remote
+{
+    /// <summary>
+    /// Helper type that abstract out JsonRpc communication with extra capability of
+    /// using raw stream to move over big chunk of data
+    /// </summary>
+    internal class JsonRpcClient : IDisposable
+    {
+        private readonly Stream _stream;
+        private readonly JsonRpc _rpc;
+
+        public JsonRpcClient(Stream stream)
+        {
+            _stream = stream;
+            _rpc = JsonRpc.Attach(stream, GetTarget());
+            _rpc.Disconnected += Disconnected;
+        }
+
+        protected virtual object GetTarget()
```

Comment:
Don't you always need a target? If so, it sounds like JsonRpcClient should be abstract and this should be an abstract method.

Or, if we want "this" to be the target by default, this method should return "this" instead of "null".